VERSION_LESS_EQUAL in CMake < 3.7 - Unknown arguments specified

I am working on building [mcrouter](https://github.com/facebook/mcrouter) and am unable to build the zstd dependency due to the following error -

```
+ git clone https://github.com/facebook/zstd
Cloning into 'zstd'...
+ cd /usr/local/mcrouter/pkgs/zstd
+ cmake -DCMAKE_INSTALL_PREFIX=/usr/local/mcrouter/install build/cmake/
CMake Error at CMakeLists.txt:18 (if):
  if given arguments:

    "3.5.1" "VERSION_LESS_EQUAL" "3.13.2"

  Unknown arguments specified


-- Configuring incomplete, errors occurred!
```

According to release notes, `VERSION_LESS_EQUAL` was not added to CMake [until 3.7](https://cmake.org/cmake/help/v3.7/release/3.7.html?highlight=version_less_equal). This change was recently made by @hjmjohnson in 97d1de3d22c5ad53a465d76a5708a4ba90d0ed8b.

Before I move forward with a fix, is the right path forward to change the [CMakeLists.txt conditional](https://github.com/facebook/zstd/blob/dev/build/cmake/CMakeLists.txt#L15-L23) to check for `${CMAKE_MINOR_VERSION}` as well? I'm not very familiar with CMake.

--- a/build/cmake/CMakeLists.txt
+++ b/build/cmake/CMakeLists.txt
@@ -12,9 +12,9 @@ cmake_minimum_required(VERSION 2.8.9 FATAL_ERROR)
 # As of 2018-12-26 ZSTD has been validated to build with cmake version 3.13.2 new policies. 
 # Set and use the newest cmake policies that are validated to work 
 set(ZSTD_MAX_VALIDATED_CMAKE_VERSION "3.13.2") 
-if("${CMAKE_MAJOR_VERSION}" LESS 3) # Cmake version 2 does not understand the VERSION_LESS_EQUAL operator
+if("${CMAKE_MAJOR_VERSION}" LESS 3) # Cmake version <3.7 does not understand the VERSION_LESS_EQUAL operator
   set(ZSTD_CMAKE_POLICY_VERSION "${CMAKE_VERSION}") 
-else()
+elseif(("${CMAKE_MAJOR_VERSION}" EQUAL 3 AND "${CMAKE_MINOR_VERSION}" GREATER 6) OR ("${CMAKE_MAJOR_VERSION}" GREATER 3))
   if("${CMAKE_VERSION}" VERSION_LESS_EQUAL "${ZSTD_MAX_VALIDATED_CMAKE_VERSION}") 
     set(ZSTD_CMAKE_POLICY_VERSION "${CMAKE_VERSION}") 
   else() 

